src/csharp/Microsoft.Spark.E2ETest/IpcTests/ML/Feature/FeatureHasherTests.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Spark.ML.Feature;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Types;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.IpcTests.ML.Feature
+{
+    [Collection("Spark E2E Tests")]
+    public class FeatureHasherTests
+    {
+        private readonly SparkSession _spark;
+        
+        public FeatureHasherTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+        }
+        
+        /// <summary>
+        /// Create a <see cref="DataFrame"/>, create a <see cref="FeatureHasher"/> and test the
+        /// available methods. Test the FeatureBase methods using <see cref="FeatureBaseTests"/>.
+        /// </summary>
+        [Fact]
+        public void TestFeatureHasher()
+        {
+            DataFrame dataFrame = _spark.CreateDataFrame(
+                new List<GenericRow>
+                {
+                    new GenericRow(new object[] {2.0D, true, "1", "foo"}),
+                    new GenericRow(new object[] {3.0D, false, "2", "bar"})
```

Comment:
Can you try to auto format? (ctrl+k and ctrl+d)?

Reply:
will do, I was trying to use rider on a mac but i'll switch to visual studio in parallels so this should work better from now on!